test/integration.test.ts

```diff
@@ -1,35 +1,206 @@
+    diffFormatLogger.close();
+    await diffFormatTransport.flush();
+
+    const event = await queryEvent(`random = '${random}'`);
+    expect(event).toBeDefined();
+    expect(getPropertyFromEvent(event, 'whats')).toBe('formats');
+  })
+
+  it('should work with no API key', async ()=>{
+    const random = getRandom();
+    const anonTransport = new winstonSeq({
+      serverUrl: process.env.SEQ_INGESTION_URL,
+      onError: (e => { console.error(e) }),
+      handleExceptions: true,
+      handleRejections: true,
+    });
+    const anonLogger = winston.createLogger({
+      level: 'debug',
+      defaultMeta: { application: 'noAPItests' },
+      transports: [
+        anonTransport,
+      ],
+    });
+
+    anonLogger.info("winston-seq: tests: should work with no API key", {random});
+    anonLogger.close();
+    await anonTransport.flush();
+
+    const event = await queryEvent(`random = '${random}'`);
+    expect(event).toBeDefined();
   })
+
+  it('should log exceptions', async ()=>{
+    const random = getRandom();
+    try{
+      throw new Error("Test error");
+    } catch (e) {
+      logger.error(e, {random});
+    }
+    await transport.flush();
+    const event = await queryEvent(`random = '${random}'`);
+    expect(event).toBeDefined();
+    expect(getPropertyFromEvent(event, 'stack').length).toBeGreaterThan(0);
+  });
+
+  it('should log all logging levels', async ()=>{
+    const random = getRandom();
+    const levels = ['error', 'warn', 'info', 'http', 'verbose', 'debug', 'silly'];
+    await Promise.all(levels.map(level => {
+        logger.log(level, "winston-seq: testing: logging levels for level {level}", {level, random, test: 'should log all logging levels'});
+        return transport.flush();
+      }));
+    const events = await queryEvents(`random = '${random}'`);
+    expect(events).toBeDefined();
+    expect(events.some((event: any) => event.Level == 'error')).toBe(true);
+    expect(events.some((event: any) => event.Level == 'silly')).toBe(true);
+    expect(events.some((event: any) => event.Level == 'http')).toBe(true);
+    expect(events.some((event: any) => event.Level == 'warn')).toBe(true);
+    expect(events.some((event: any) => event.Level == 'info')).toBe(true);
+    expect(events.some((event: any) => event.Level == 'verbose')).toBe(true);
+    expect(events.some((event: any) => event.Level == 'debug')).toBe(true);
+  });
 })
+
+function getPropertyFromEvent(event: any, propertyName: string) {
+  return event.Properties.find((p: any) => p.Name === propertyName).Value;
+}
+
+function getMessageFromEvent(event: any) {
+  return event.MessageTemplateTokens[0].Text;
+}
+
+function getRandom() {
+  return Math.round(Math.random() * 1000000).toString();
+}
+
+async function queryEvent(filter: string) {
+  const response = await axios.get(`${process.env.SEQ_API_URL}/api/events/signal?filter=${encodeURIComponent(filter)}&count=1&shortCircuitAfter=100&apiKey=${process.env.SEQ_API_KEY}`);
```

Comment:
Can add `&render` to these URLs to avoid having to work with the individual message template tokens in responses.

Reply:
FWIW it's also frequently nicer to request newline-delimited CLEF with `&clef`, but that will make parsing harder here.